Mathlib/MeasureTheory/Integral/Marginal.lean

```diff
@@ -0,0 +1,243 @@
+/-
+Copyright (c) 2023 Floris van Doorn. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Floris van Doorn, Heather Macbeth
+-/
+import Mathlib.MeasureTheory.Constructions.Pi
+import Mathlib.MeasureTheory.Integral.Lebesgue
+
+/-!
+# Marginals of multivariate functions
+
+In this file, we define a convenient way to compute integrals of multivariate functions, especially
+if you want to write expressions where you integrate only over some of the variables that the
+function depends on. This is common in induction arguments involving integrals of multivariate
+functions.
+This constructions allows working with iterated integrals and applying Tonelli's theorem
+and Fubini's theorem, without using measurable equivalences by changing the representation of your
+space (e.g. `((ι ⊕ ι') → ℝ) ≃ (ι → ℝ) × (ι' → ℝ)`).
+
+## Main Definitions
+* If `f : α → ℝ≥0∞` then `lmarginal μ s f` or `∫⋯∫⁻_s, f ∂μ` is the function that integrates `f`
```

Comment:
Could you clarify that `α` is not a generic type here: it has to be a product type for the construction to make sense. And that the measure is not a generic measure either: it has to be a product measure (more precisely, mu is not a measure on the product space, but a family of measure on each factor, right?)

Note that the notion of marginal in probability theory is also very important for measures which are not product measures. For your goal here (having a condition version of Fubini), assuming that the measure is a product measure is completely reasonable, but you should emphasize this prominently in the file-level docstring and also in the docstring of the definition.